more/pixels/pixelConfig.js

```diff
@@ -0,0 +1,183 @@
+import harden from '@agoric/harden';
+
+import { makePixelMintKeeper } from './pixelMintKeeper';
+import { makePixelListAssayMaker } from './pixelAssays';
+
+/**
+ * `makePixelConfigMaker` exists in order to pass in more parameters
+// than makePixelConfig allows.
+ * @param  {function} makeUseObj creates a "use object", which has all
+ * of the non-ERTP methods for assets that are designed to be used.
+ * For instance, a pixel can be colored. The use object is associated
+ * with an underlying asset that provides the authority to use it. 
+ * @param  {number} canvasSize=10 the size of the gallery in pixel
+ * squares across and down
+ * @param  {issuer} parentIssuer the `parentIssuer` is used when
+ * creating a revocable childPayment or childPurse, as in the
+ * landowner/tenant/subtenant pattern. In that pattern, the owner
+ * holds assets associated with the parent issuer, the tenant holds
+ * assets associated with the child issuer, the subtenant holds assets
+ * associated with the grandchild issuer, and so forth.
+ */
+function makePixelConfigMaker(makeUseObj, canvasSize = 10, parentIssuer) {
+  function makePixelConfig(makeMint, description) {
+    // The childIssuer/childMint are lazily created to avoid going
+    // infinitely far down the chain of issuers on creation. The
+    // childIssuer assets are revocable by the current issuer (the
+    // child's parent).
+    let childIssuer;
+    let childMint;
+
+    const makePixelAssay = makePixelListAssayMaker(canvasSize);
+
+    // Lazily creates the childMint if it doesn't already
+    // exist
+    function getOrMakeChildMint(issuer) {
+      if (childMint === undefined) {
+        const makePixelConfigChild = makePixelConfigMaker(
+          makeUseObj,
+          canvasSize,
+          issuer,
+        );
+        childMint = makeMint(description, makePixelConfigChild);
+        childIssuer = childMint.getIssuer();
+      }
+      return childMint;
+    }
+
+    // This method is used in the creation of childPayments and
+    // childPurses where we want the amount to be the same as in the
+    // original asset apart from the difference in issuers.
+    function getChildAmount(issuer, amount) {
+      getOrMakeChildMint(issuer);
+      // quantity is the same, but amounts are different for
+      // different issuers
+      const { quantity } = amount;
+      return childIssuer.makeAmount(quantity);
+    }
+
+    return harden({
+      makeCustomPayment(superPayment, issuer) {
+        return harden({
+          ...superPayment,
+          // This creates a new use object on every call. Please see
+          // the gallery for the definition of the use object that is
+          // created here by calling `makeUseObj`
+          getUse() {
+            return makeUseObj(issuer, superPayment);
+          },
+          // Mint a new payment from the child mint with the same
+          // quantity as the original payment
+          getChildPayment() {
```

Comment:
Given payment `p` at level 1 with balance B, does `p.getChildPayment()` mint a new child payment at level 2 with balance `B` each time `getChildPayment` is called? Does this violate an intended conservation property?

Reply:
Oh gosh, thanks so much for catching this. I've changed it to check if a childPayment is created before creating one. After creation, the balances of the parent and child payment are entirely separate. If the balance of the parent changes and the user wants to create a new childPayment, they should create a new parent purse or payment and derive a new childPayment associated with it.